aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -278,6 +278,18 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    /// Entry function-only rotation key function that allows the signer update their authentication_key.
+    entry fun rotate_authentication_key_call(account: &signer, new_auth_key: vector<u8>) acquires Account, OriginatingAddress {
```

Comment:
Not the greatest of names, but I guess this works.

Might want to put a warning that the user should ensure they have access to the private key mapping to the authentication key, or they'll lose all access.